### PR TITLE
feature/dependency-review-1

### PR DESCRIPTION
**Pull request by lc-chrisbarton**
updated body

### CODE REVIEW

File: .github/workflows/dependency_review.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+          set -o errexit
+          TMP_GIT_STATUS_GO_MOD=$(git diff go.mod)
+          echo "TMP_GIT_STATUS_GO_MOD=${TMP_GIT_STATUS_GO_MOD}"
+
+          if [ "${TMP_GIT_STATUS_GO_MOD}" == "" ]; then
+            echo "Dependencies are NOT outdated"
+
+            else
+              echo "Dependencies are outdated"
+              exit 1
```
Comment (lc-caigwatkin) [inline .github/workflows/dependency_review.yaml]: could create a [warning annotation](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-a-warning-message) in `go.mod` when deps are outdated but still `exit 0` from the action

**Review by lc-caigwatkin — DISMISSED** [review]

File: .github/workflows/dependency_review.yaml (hunk 1 above)
Comment (lc-chrisbarton) [reply .github/workflows/dependency_review.yaml]: Of limited use, see wip 14

Comment (lc-caigwatkin) [reply .github/workflows/dependency_review.yaml]: it looks like annotations can be made to show up on a specific line in a file: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks#checks

Comment (lc-caigwatkin) [reply .github/workflows/dependency_review.yaml]: https://docs.github.com/en/rest/checks/runs#create-a-check-run

**Review by lc-chrisbarton — COMMENTED** [review]
foo

**Pull request closed without merging**